enhancements/oc/cli-manager.md

```diff
@@ -0,0 +1,252 @@
+follow. Currently. the `cli-artifacts` image stream in the `openshift namespace` is shipped with the release payload. This is where the artifacts
+for `oc` will be extracted from. If other CLIs choose to offer artifacts in this way, they can opt in by supplying an image to be
+pulled from their desired image registry.
+
+_For this proposal "ClusterCLI" is assumed the name of a CR that will store metadata and information about each CLI_
+
+## Motivation
+
+As more services are created on top of OpenShift, more CLIs are introduced to simplify interaction with these services.
+Some current examples are oc, kubectl, odo, istio, tekton, and knative.  It is difficult for users to discover what tools exist,
+where to download them from and which version they should download.  We need to simplify as much as possible the interaction
+of services on OpenShift.  We need a mechanism for providing and consuming tools that is simple to add on to as new CLIs are
+developed from a variety of sources - and this should be specific for each cluster and available with disconnected installs.
+
+### Goals
+
+* No new form of binary distribution or binary creation will be proposed, because we have an existing structure at Red Hat.
+RPMs or images are the only options, and images must be deployed by the RH pipeline via operators. This proposal is for delivering
+CLIs via images, because this will enable offering CLIs offline through an existing image registry.
+* CLI owners must be able to easily distribute their binaries
+* The version of the CLI a user is offered is appropriate for the version of the CLI controller installed on the cluster
+* Arbitrary binaries not delivered by the in-cluster CRD controller are not important or relevant because of requirement 1 and 2
+  * Anyone who wants such tools can download them outside of this mechanism
+* CRD controller for registering CRs with an API for listing, extracting and downloading artifacts
+  * Krew-compatible index that both krew and oc can consume where the download links are for cluster-local resources
+  * The index and artifcats will served from a single route and service within the cluster
+
+### Non-Goals
+
+* `oc tools` will not build or serve the binaries.  It will know where to find them.
+* This proposal is not concerned with _which_ binaries will be managed.  This proposal is meant to determine the mechanism only.  Consumers and publishers are clients of the mechanism.
+* `oc tools` will not create or update the ClusterCLI Custom Resources, that will be managed by individual CLIs. However, the CLIs that are
+currently available as `consoleclidownloads`, `oc, odo, and helm`, will be included in the reference implementation of this enhancement.
+
+## Proposal
+
+Each component wishing to provide customers with their binaries will build and publish artifacts via an official channel to a central index.
```

Comment:
This is a good non-goal, because we don't want to deal with how binaries are shipped through images. We only care about the end result, ie. the image.

Reply:
That looks like a holdover from the original proposal, removed and clarified